tests/net/lib/lwm2m/functional/pytest/test_lwm2m.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+import logging
+import pytest  # noqa # pylint: disable=unused-import
+from leshan import Leshan
+import re
+import os
+import binascii
+from coapthon.client.helperclient import HelperClient
+import random
+import string
+
+from twister_harness.device.device_abstract import DeviceAbstract
+
+logger = logging.getLogger(__name__)
+
+class Shell(object):
+    def __init__(self, dut) -> None:
```

Comment:
You may use annotations (`dut: DeviceAbstract`). It will be much easier to read/develop the code with some IDE